openedx_events/learning/data.py

```diff
@@ -257,3 +257,42 @@ class UserNotificationData:
     app_name = attr.ib(type=str)
     course_key = attr.ib(type=CourseKey)
     context = attr.ib(type=dict, factory=dict)
+
+
+@attr.s(frozen=True)
+class ProgramData:
+    """
+    Attributes defined for the Open edX Program data object.
+
+    Arguments:
+        uuid (str): The UUID of the program (from Course-Discovery)
+        title (str): The title of the program
+        program_type (str): The type slug of the program (e.g. professional, microbachelors, micromasters, etc.)
+    """
+
+    uuid = attr.ib(type=str)
+    title = attr.ib(type=str)
+    program_type = attr.ib(type=str)
+
+
+@attr.s(frozen=True)
+class ProgramCertificateData:
+    """
+    Attributes defined for the Open edX Program Certificate data object.
+
+    Arguments:
+        user (UserData): User associated with the Program Certificate
+        program (ProgramData): Program data associated with the Program Certificate
+        uuid (str): UUID of the UserCredential record in Credentials
+        certificate_available_date (datetime): Optional. A DateTime describing when a leaner is allowed to view the
+                                                credential
+        status (str): The status of the credential (e.g. `awarded` or `revoked`)
```

Comment:
iirc we define this set of supported statuses somewhere in Credentials. Do we have any way of referencing that in comments like this?

Reply:
In Credentials we only have two statuses -- `awarded` or `revoked`. Course Certificates in the LMS support many more statuses. 

https://github.com/openedx/credentials/blob/7a08a267605a3d1e913ec81a3566609ffd09ac8f/credentials/apps/credentials/models.py#L158-L161.

I can add a link to this if you think it's useful!